skater/gameboard.py

```diff
@@ -39,4 +37,25 @@ def obstacles_left(self, player):
             obstacle
             for obstacle in self.obstacles
             if obstacle.is_to_the_left(player.rect)]  # see `obstacles_right`
-        return ans
\ No newline at end of file
+        return ans
+
+    def limit_under(self, player):
+        limits = [
+            obstacle.rect.top
+            for obstacle in self.obstacles_under(player)]
+        
+        return min(limits + [self.MAX_POSITION])
+
+    def limit_right(self, player):
+        limits = [
+            obstacle.rect.left
+            for obstacle in self.obstacles_right(player)]
+
+        return min(limits + [self.MAX_POSITION])
```

Comment:
to jest po to, zeby sie nie wywalalo, jak nie ma zadnej przeszkody w liscie, tak?

Reply:
tzn te max_poz. czy 2*31 to jest max dla tego typu?

Reply:
Tak, to jest po to zeby się nie wywalało.

Wartość 2*31 jest w sumie tak znikąd, po prostu duża liczba.